packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc

```diff
@@ -0,0 +1,170 @@
+#include <flutter/event_sink.h>
+#include <flutter/event_stream_handler_functions.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar.h>
+#include <flutter/standard_method_codec.h>
+#include <net_connection.h>
+#include <wifi-manager.h>
+
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "log.h"
+
+class ConnectivityPlusTizenPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar) {
+    LOG_INFO("RegisterWithRegistrar");
+    auto plugin = std::make_unique<ConnectivityPlusTizenPlugin>();
+    plugin->SetupChannels(registrar);
+    registrar->AddPlugin(std::move(plugin));
+  }
+
+  ConnectivityPlusTizenPlugin() : m_connection(nullptr), m_events(nullptr) {
+    ensureConnectionHandle();
+  }
+
+  virtual ~ConnectivityPlusTizenPlugin() {
+    if (m_connection != nullptr) {
+      connection_destroy(m_connection);
+      m_connection = nullptr;
+    }
+  }
+
+  void registerObsever(
+      std::unique_ptr<flutter::EventSink<flutter::EncodableValue>> &&events) {
+    ensureConnectionHandle();
+    if (connection_set_type_changed_cb(m_connection, connetionTypeChangedCB,
+                                       this) != CONNECTION_ERROR_NONE) {
+      return;
+    }
+    m_events = std::move(events);
+  }
+
+  void clearObserver() {
+    if (m_connection == nullptr || m_events == nullptr) return;
+
+    connection_unset_type_changed_cb(m_connection);
+    m_events = nullptr;
+  }
+
+  void sendConnectivityChangedEvent(connection_type_e state) {
+    if (m_events == nullptr) return;
+    std::string replay = convertConnectionTypeToString(state);
+    flutter::EncodableValue msg(replay);
+    m_events->Success(msg);
+  }
+
+ private:
+  static void connetionTypeChangedCB(connection_type_e state, void *data) {
+    LOG_DEBUG("connetionTypeChangedCB");
+    ConnectivityPlusTizenPlugin *plugin_pointer =
+        (ConnectivityPlusTizenPlugin *)data;
+    plugin_pointer->sendConnectivityChangedEvent(state);
+  }
+
+  void ensureConnectionHandle() {
+    if (m_connection == nullptr) {
+      if (connection_create(&m_connection) != CONNECTION_ERROR_NONE) {
+        m_connection = nullptr;
+      }
+    }
+  }
+
+  std::string convertConnectionTypeToString(connection_type_e net_state) {
+    std::string result;
+    switch (net_state) {
+      case CONNECTION_TYPE_WIFI:
+      case CONNECTION_TYPE_ETHERNET:
+        result = "wifi";
+        break;
+      case CONNECTION_TYPE_CELLULAR:
+        result = "mobile";
+        break;
+      case CONNECTION_TYPE_DISCONNECTED:
+      default:
+        result = "none";
+    }
+    return result;
+  }
+
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    ensureConnectionHandle();
+    LOG_INFO("method : %s", method_call.method_name().data());
```

Comment:
Should we still care about [this](https://github.com/flutter-tizen/flutter-tizen/issues/71)?

Reply:
I believe statically linking higher version libstdc++ by flutter-tizen/flutter-tizen/pull/106 solved the issue.

Reply:
Oh you're right! I forgot this :smiley: